### PR TITLE
Parser 'number' that advances by column first

### PR DESCRIPTION
**Pull request by yoavram**
Great package!
I've added a parser 'number_col' that works like 'number' but advances by column first.

### CODE REVIEW

**Comment by eyurtsev** [comment]
I am finally finishing the migration of the repository to git from bitbucket. Thanks for the pull request! I'll resume activity soon on development (finally done PhDing :)

File: FlowCytometryTools/core/bases.py [hunk 1]
```diff
@@ -841,9 +842,15 @@ def _get_ID2position_mapper(self, parser):
...
+        elif parser.startswith('number'):
+            by_col = parser.endswith("col")
             def num_parser(x):
-                i,j = unravel_index(int(x-1), self.shape, order='F')
+                w,h = self.shape
+                if by_col:
+                    w,h = h,w
+                i,j = unravel_index(int(x-1), (w,h), order='F')
+                if by_col:
+                    i,j = j,i
```
Comment (eyurtsev) [inline FlowCytometryTools/core/bases.py]: I think this should work
```
order = 'C' if by_col else 'F'
 i,j = unravel_index(int(x-1), self.shape, order=order)
```

File: FlowCytometryTools/core/bases.py [hunk 2]
```diff
@@ -841,9 +842,15 @@ def _get_ID2position_mapper(self, parser):
             parser = lambda x: parser[x]
         elif parser == 'name':
             parser = lambda x: (x[0], int(x[1:]))
-        elif parser == 'number':
+        elif parser.startswith('number'):
```
Comment (eyurtsev) [inline FlowCytometryTools/core/bases.py]: how about calling it 'number_col_first' to make it more explicit that it iterates by columns first? I can rename the other one to be 'number_row_first'.

**Comment by eyurtsev** [comment]
I left a few comments in the source code, but don't worry about updating the pull request. I'll integrate it with a few additional changes. (Mainly I need to clean up the code since it's a mess now.)

**Comment by yoavram** [comment]
No problem, rename sounds good, I didn't know what to call it myself ("There are only two hard things in Computer Science: cache invalidation and naming things")

**Pull request merged**